config/application.rb

```diff
@@ -80,6 +80,36 @@ class Application < Rails::Application
       **config.active_record.encryption,
     )
 
+    # Don't implicitly invert plural associations. This causes some "correct"
+    # but unexpected behavior. E.g. given these models:
+    #
+    #   product = Product.new
+    #   release = Release.new(product:)
+    #
+    # The following returns a collection including the initialized release,
+    # even if we eventually throw out that release (e.g. during tests):
+    #
+    #   product.releases
+    #
+    # In addition, when the product is saved, so are all of its implicitly
+    # associated releases, which is unexpected:
+    #
+    #   product.save
+    #
+    # I prefer to keep this explicit:
+    #
+    #   product = Product.new(
+    #     releases: [Release.new(product:)],
+    #   )
+    #
+    config.active_record.automatically_invert_plural_associations = false
+
+    # Implicitly define inverses for has-many associations
+    config.active_record.has_many_inversing = true
```

Comment:
Took 3 days to fully figure this out. See above comment for full explanation of the "unexpected" behavior. I understand it's "correct" i.r.t. inverses, but — to me at least — it doesn't make sense to save associated records implicitly. What if one of those records were discarded, e.g. a default that is later replaced by a new record during initialization? May seem like a contrived example, but that happens all the time in factory-based tests.